### PR TITLE
Separate name

### PR DESCRIPTION
**Pull request by GeoffThorpeFT**

### CODE REVIEW

**Comment by GeoffThorpeFT** [comment]
dev url is here, if you want to play with the result of my changes. http://oi-endpoint-manager-dev.herokuapp.com.  I dont think I've covered all the interconnectivity between the fields

File: views/endpoint.ms [hunk 1]
```diff
@@ -215,12 +218,17 @@
 
 						</aside>
 					</div>{{/saved}}
+					<div class="o-forms-group">
+						<label class="o-forms-label" for="name">Base URL</label>
```
Comment (GeoffThorpeFT) [inline views/endpoint.ms]: I agree.
Was that the only comment? wow!

File: index.js [hunk 2]
```diff
@@ -195,27 +197,28 @@ function endpointController(endpoint) {
 	var protocols = [];
 	if (['http', 'https'].indexOf(endpoint.protocol) != -1) {
 		protocols.push(endpoint.protocol);
-		endpoint.baseurl = endpoint.protocol+"://"+endpoint.id+"/";
+		endpoint.baseurl = endpoint.protocol+"://"+endpoint.baseURL+"/";
```
Comment (lucas42) [inline index.js]: I'm not sure how mustache handles the difference between `baseurl` and `baseURL`, but just to be safe (and to help the sanity of anyone who looks at this code in future), it'd probably be prudent to call these something different and not rely on case to distinguish between them.

**Comment by lucas42** [comment]
Existing endpoints don't have baseURL set.  Either we should go through and set those, or handle that case specifically by defaulting to ID.

File: views/endpoint.ms [hunk 3]
```diff
@@ -235,14 +243,14 @@
 						<label class="o-forms-label" for="healthSuffix">Healthcheck Suffix</label>
 						<small class="o-forms-additional-info">The path of a healthcheck which meets our <a href="https://docs.google.com/document/d/18hefJjImF5IFp9WvPAm9Iq5_GmWzI9ahlKSzShpQl1s/edit">Healthcheck Standard</a></small>
 						<div class="o-forms-affix-wrapper">
-							<span class="o-forms-prefix baseurl">{{baseurl}}</span>
+							<span class="o-forms-prefix base">{{base}}</span>
```
Comment (lucas42) [inline views/endpoint.ms]: This line shouldn't change.  It should be `baseurl` (both in the class and the mustache variable)

File: views/endpoint.ms [hunk 4]
```diff
@@ -235,14 +243,14 @@
...
 							<input name="healthSuffix" value="{{healthSuffix}}" default="__health" class="o-forms-text" id="healthSuffix"></input>
 						</div>
 					</div>
 					<div class="o-forms-group">
 						<label class="o-forms-label" for="aboutSuffix">About Suffix</label>
 						<div class="o-forms-affix-wrapper">
-							<span class="o-forms-prefix baseurl">{{baseurl}}</span>
+							<span class="o-forms-prefix base">{{base}}</span>
```
Comment (lucas42) [inline views/endpoint.ms]: This line shouldn't change.  It should be `baseurl` (both in the class and the mustache variable)

File: views/endpoint.ms [hunk 5]
```diff
@@ -141,11 +141,14 @@
 					var idinput = document.querySelector('#id');
 					addChecks(idinput, idinput.parentNode);
 
+					var baseinput = document.querySelector('#base');
```
Comment (lucas42) [inline views/endpoint.ms]: When this changes, we'll want to update what's shown next to the suffix fields.  You should be able to add:
```
baseinput.addEventListener('change', updateBaseURL);
```
next to the similar lines for idinput and protocolinput below.

**Pull request merged**